ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/XARecoveryModuleUnitTest.java

```diff
@@ -400,16 +400,24 @@ public void start(Xid xid, int i) throws XAException {
             }
         });
 
+        int oldCount = recoverCalled;
+        callFirstPass(xarm, oldCount);
```

Comment:
We have lost some test precision here, before we asserted the actual number of recover calls but now we are just testing that the number is incrementing on the first path and I'd expect the test would fail if you called callSecondPass twice in a row. Since the expectation on the error free path is two recover calls on the first pass and no recover calls on the second pass, the following would be more maintainable:

```
        int oldCount = recoverCalled;
        callFirstPass(xarm, oldCount + 2);
        callSecondPass(xarm, oldCount + 2);
        callFirstPass(xarm, oldCount + 4);
        callFirstPass(xarm, oldCount + 6);
        callSecondPass(xarm, oldCount + 6);
```

or better still, because the following makes it obvious what the test is doing:

```
        int oldCount = recoverCalled;
        // on the error free path
        // we expect periodicWorkFirstPass to make two recover calls,
        // and we expect periodicWorkSecondPass to make no recover calls
        xarm.periodicWorkFirstPass();
        assertEquals(recoverCalled, oldCount + 2);
        xarm.periodicWorkSecondPass();
        assertEquals(recoverCalled, oldCount + 2);
        xarm.periodicWorkFirstPass();
        assertEquals(recoverCalled, oldCount + 4);
        xarm.periodicWorkFirstPass();
        assertEquals(recoverCalled, oldCount + 6);
        xarm.periodicWorkSecondPass();
        assertEquals(recoverCalled, oldCount + 6);
```